documentation/specs/proposed/General_perf_onepager.md

```diff
@@ -0,0 +1,42 @@
+# General performance improvements
+MSBuild is the main tool used for building .NET projects. It is used by Visual Studio, the .NET CLI, and other build tools. The performance of MSBuild is critical to the productivity of developers. This document outlines our goals to improve overall performance of MSBuild.
+
+## Goals and Motivation
+
+We are aiming for:
+ - Searching opportunities to improve the build times.
+ - Reducing MSBuild overhead in the Visual Studio IDE.
```

Comment:
I would pick a strawman, such as "Reduce Visual Studio rebuild time at @ 75th by 20%.". We already have telemetry and tracking of this on our performance dashboards on ShipReady. Unlike almost every single one of performance metrics, release over release, rebuild has gotten worse.

Reply:
It's difficult right now to set a target number when we don't know exactly what is possible to achieve within this scope. I would keep it as a best-effort task, and we can sync on selecting particular improvements based on priorities, such as UI delay, rebuild time, etc.